### ISSUE TITLE
download both audio and video of same source 

### ISSUE THREAD

**Issue by binarygeek119** [meeb/tubesync#101]
i would like to download both audi and video from some sources

for example
epic rap battles of history channel 
https://www.youtube.com/user/ERB
as i want both the audio only and the video aswell

**Comment by meeb** [comment]
Hi. I can see why you would want this but this is probably a bit challenging (either technically, or via messy UI) to implement in TubeSync in its current state. I'll put this onto the future potential wishlist, but if this is implemented it won't be for a while. Thanks for the feedback and suggestion!

**Comment by PhuriousGeorge** [comment]
I believe this is similar enough to #91 youtube-dl cannot download video and audio simultaneously, so imho there are two reasonable options:
1. Adding two identical sources with different settings 
2. Download video and post-process to extract audio stream via ffmpeg

**Comment by meeb** [comment]
Yep, you are correct @PhuriousGeorge (this was the "either technically, or via messy UI") bit in my original comment. Adding duplicate media to sources would require significant work as that's currently a database constraint relied on in a bunch of places and I'd really rather not implement subprocess post-process calling of ffmpeg in TubeSync if it can be avoided as this would also complicate downloads where one download had to rely on another download completing first.

While I can see the point of the feature, this may well be on the wishlist for a while!

**Comment by csjoh** [comment]
I would like to see this feature as well, but as a workaround:

1. Spin up a second container with its own volume and its own settings.
2. Add the channel source you want both audio and video from separately, and adjust settings to your needs.

The download location could conceivably be the same as the first container, but I wouldn't risk having the two instances overwrite each others' downloaded files. They would definitely need separate config locations.

**Comment by meeb** [comment]
Running a second TubeSync instance is probably likely to be the only sensible way to handle this for the time being. I can't see the required internal restructuring to allow for multiple copies of the same media to be stored correctly with different source settings being added any time soon. Thanks for the suggestion for others.

**Comment by sywhn** [comment]
Hello, there is a‘‘--keep-video’’ option in the yt-dlp, is it possible to add this option in tubesync? It will keep both a video-only file and an audio-only file and it will be best we can choose to keep the video-only file or the audio-only file. Thank you. 

--keep-video                Keep the intermediate video file on disk after post-processing

**Comment by meeb** [comment]
That's probably a different issue @sywhn - also what would be the use case for this feature over just audio-only? Is there actually a requirement for video-only downloads?

**Comment by meeb** [comment]
Ah I see, sounds sensible. The default `yt-dlp` flags are editable in the `settings.py` (or override the dict in `local_settings.py`):

https://github.com/meeb/tubesync/blob/main/tubesync/tubesync/settings.py#L157

I'll see how feasible it is to make this a config flag. Note these are the API parameters, not the CLI arguments.

**Comment by sywhn** [comment]
Thank you. Look forward to it.

**Comment by tcely** [comment]
Adding `'keepvideo': True,` would seem to be the best option.

Adding a exec post processor command to delete the video only file and keep the audio only and combined files would be my next improvement.
